src/manifest.c

```diff
@@ -776,24 +779,49 @@ static int write_manifest_plain(struct manifest *manifest)
 	}
 
 	list = g_list_first(manifest->manifests);
+	string_or_die(&manifest_tempdir, "./XXXXXX");
```

Comment:
By state directory, I mean the configurable state directory (defaults to /var/lib/update). I simply think it's better for swupd-server to always be writing in the state directory whenever possible. One analogous example for using temporary directories is pack content being staged in STATE_DIR/packstage, and then removed after the packs are created. You're right that swupd-server *could* write in the current directory, or in /tmp, etc. but the state directory is IMO the best place the keep temporary state.